### PR TITLE
Split css server templates

### LINKED ISSUE #8126: Split out the CSS for default templates
This issue comes from PR #8105 and is also related to TryGhost/Ghost-Admin#574.

In the Ghost server-side code we have 3 "default" templates, which provide some basic styles for a special page, and are allowed to be overridden by a theme.

At the moment, all 3 of these default templates get their styles by importing the ghost.css stylesheet from Ghost-Admin. However, we're trying very hard to separate out the server-side theme part of Ghost and the admin panel, and so we'd like those templates to have their own styles, which can be loaded separately.

To make it possible for us to not load the admin css for these templates, the CSS required to display these templates needs to be split out into a separate file, which can live in shared/ for now.

----

The 3 templates are:

### PR DESCRIPTION
**Pull request by aileen**
closes #8126
needs e3acd3c

This is a replacement PR of #8217 (thanks @TienSFU25 for the work 🤗), because these changes are needed urgently and blocking other work.

Adds a new `ghost.css` file in `/core/shared/` to be used for server side template rendering (`error.hbs`, `subscribe.hbs` and `private.hbs`).

### CODE REVIEW

File: core/server/apps/private-blogging/lib/views/private.hbs [hunk 1]
```diff
@@ -15,7 +15,7 @@
         <link rel="shortcut icon" href="{{asset "favicon.ico"}}">
         <meta http-equiv="cleartype" content="on">
 
-        <link rel="stylesheet" href="{{asset "ghost.css" ghost="true" minifyInProduction="true"}}" />
```
Comment (kirrg001) [inline core/server/apps/private-blogging/lib/views/private.hbs]: Can `ghost` and `minifyInProduction` be removed from `core/helpers/asset`?

Comment (AileenCGN) [inline core/server/apps/private-blogging/lib/views/private.hbs]: It thought that was the plan?

Comment (kirrg001) [inline core/server/apps/private-blogging/lib/views/private.hbs]: Just thought we can make this change already in this PR. But i am fine with having a separated PR 👍

Comment (ErisDS) [inline core/server/apps/private-blogging/lib/views/private.hbs]: I don't think the asset helper can be removed here? 

Has this PR been tested on a subdirectory setup?

Comment (AileenCGN) [inline core/server/apps/private-blogging/lib/views/private.hbs]: You're right! Does not work with subdirectory setup. Good catch! I will add the `{{asset}}` helper back.

**Comment by AileenCGN** [comment]
I added some logic to `getAssetUrl()`, to not add `/assets/` to the URL, when the `ghost.css` stylesheet is requested. BUT, I'm not sure if this is the right approach...

**Comment by ErisDS** [comment]
I don't think there was any need to add extra logic here. This is a shared file, so the path passed to the asset helper should be `/shared/ghost.css` - the line of code in getAssetUrl() that was edited already has logic for this.

**Comment by AileenCGN** [comment]
This is ready for review and merge 🎉

**Review by kirrg001 — APPROVED** [review]
This LTGM now 👍

**Pull request merged**